spec/immutable_struct_spec.rb

```diff
@@ -103,14 +104,61 @@ def self.from_array(array)
   end
 
   describe "to_h" do
-    it "should include the output of params and block methods in the hash" do
-      klass = ImmutableStruct.new(:flappy) do
-        def lawsuit
-          'pending'
+    context "vanilla struct with just derived values" do
+      it "should include the output of params and block methods in the hash" do
+        klass = ImmutableStruct.new(:name, :minor?, :location, [:aliases]) do
+          def nick_name
+            'bob'
+          end
         end
+        instance = klass.new(name: "Rudy", minor: "ayup", aliases: [ "Rudyard", "Roozoola" ])
+        instance.to_h.should == {
+          name: "Rudy",
+          minor: "ayup",
+          minor?: true,
+          location: nil,
+          aliases: [ "Rudyard", "Roozoola"],
+          nick_name: "bob",
+        }
+      end
+    end
+    context "additional method that takes arguments" do
+      it "should not call the additional method" do
+        klass = ImmutableStruct.new(:name, :minor?, :location, [:aliases]) do
+          def nick_name
+            'bob'
+          end
+          def location_near?(other_location)
+            false
+          end
+        end
+        instance = klass.new(name: "Rudy", minor: "ayup", aliases: [ "Rudyard", "Roozoola" ])
+        instance.to_h.should == {
+          name: "Rudy",
+          minor: "ayup",
+          minor?: true,
+          location: nil,
+          aliases: [ "Rudyard", "Roozoola"],
+          nick_name: "bob",
+        }
+      end
+    end
+
+    context "no-arg method that uses to_h" do
+      it "blows up" do
+        klass = ImmutableStruct.new(:name, :minor?, :location, [:aliases]) do
+          def nick_name
+            'bob'
+          end
+          def to_json
+            to_h.to_json
+          end
+        end
+        instance = klass.new(name: "Rudy", minor: "ayup", aliases: [ "Rudyard", "Roozoola" ])
+        expect {
```

Comment:
I don't know if this is fixable without either some way to "flag" methods as "don't include in serialization" or some version of #5.

I feel like the use-case you want generally is "serialize everything but this method:" but I'm not 100% sure.